### PR TITLE
fix: handle changed oneofs when updating wire set fields

### PR DESCRIPTION
**Pull request by ericwenn**
When which oneof field is different between src and dst, it should
always be updated. Prevents panics when oneofs contain messages and one
of their fields are set.

Still needs to be handled when a field mask with paths pointing to a
oneof field is provided.

### CODE REVIEW

File: fieldmask/update.go [hunk 1]
```diff
@@ -46,6 +46,16 @@ func Update(mask *fieldmaskpb.FieldMask, dst, src proto.Message) {
...
+	oneofs := dst.Descriptor().Oneofs()
+	for i := 0; i < oneofs.Len(); i++ {
+		oneof := oneofs.Get(i)
+		whichDst := dst.WhichOneof(oneof)
+		whichSrc := src.WhichOneof(oneof)
+		if whichDst != whichSrc && whichSrc != nil {
+			dst.Set(whichSrc, src.Get(whichSrc))
+		}
+	}
 	src.Range(func(field protoreflect.FieldDescriptor, value protoreflect.Value) bool {
```
Comment (odsod) [inline fieldmask/update.go:59]: do we need to account for oneof fields down here too? i.e. ship oneof fields that were handled in the code above?

Comment (ericwenn) [reply fieldmask/update.go:59]: It's is not needed. It's essentially a noop in the `src.Range` block as the fields have already been updated, and I'd argue that the overhead of keeping track of it is not worth it.

Comment (ericwenn) [reply fieldmask/update.go:59]: As we discussed, I'm planning to add a mother-of-all proto messages in this repository so we can get proper test coverage for more edgy cases like this.

Comment (odsod) [reply fieldmask/update.go:59]: Makes sense, I get it.

**Review by odsod — APPROVED** [review]
:+1:

**Comment by ericwenn** [comment]
superseded by #39

**Pull request closed without merging**